### PR TITLE
Feat/mesh-panel

### PR DESCRIPTION
**Pull request by dawidsdh**

### CODE REVIEW

File: src/facade/index.ts [hunk 1]
```diff
@@ -290,14 +291,11 @@ export const cherryFacade = (cherryViewer: CherryViewer) => {
...
-    propertyName: GetterSetterPropertyType,
-    value: ProjectManagerObjectPropertyType
-  ) => {
+  const setObjectProperty = (key: CherryKey, propertyName: any, value: any) => {
     try {
-      const object = pm.getObject(key);
-      (object[propertyName] as ProjectManagerObjectPropertyType) = value;
+      const object = pm.getObject(key) as any;
+      object[propertyName] = value;
+      console.log(key, propertyName, value, object);
```
Comment (damian08kk) [inline src/facade/index.ts:298]: console

File: src/facade/index.ts [hunk 2]
```diff
@@ -503,58 +523,62 @@ export const cherryFacade = (cherryViewer: CherryViewer) => {
...
-            break;
+    try {
+      const parentObject = parent ? pm.getObject(parent.key) : null;
+      const currentObject = pm.addObject(node, entities, parentObject, key);
+      const currentEntity = entities[node.key];
+
+      if (node.type === NODE_TYPES.light) {
+        const { key } = node;
+        const light = pm.getObject(key);
+        let lightobj = scene.getObject(key);
```
Comment (damian08kk) [inline src/facade/index.ts:534]: Maybe `lightObj`?

File: src/facade/index.ts [hunk 3]
```diff
@@ -503,58 +523,62 @@ export const cherryFacade = (cherryViewer: CherryViewer) => {
...
+            light.finalPosition[2]
+          );
+          const scale = vec3.fromValues(1 / 100, 1 / 100, 1 / 100);
+          const m = mat4.create();
+          mat4.translate(m, m, vec);
+          mat4.scale(m, m, scale);
+          lightobj.setTransformMatrix(m);
+        };
+
+        // light.clearChangeHandlers();
```
Comment (damian08kk) [inline src/facade/index.ts:556]: commented line

File: src/facade/index.ts [hunk 4]
```diff
@@ -503,58 +523,62 @@ export const cherryFacade = (cherryViewer: CherryViewer) => {
...
+        light.addChangeListener((type: string) => {
```
Comment (damian08kk) [inline src/facade/index.ts:557]: Doesn't we have a type for `type`? `UpdateTypes`?

**Review by damian08kk — APPROVED** [review]

File: src/facade/index.ts (hunk 4 above)
Comment (dawidsdh) [reply src/facade/index.ts:557]: No, this is the property name and temporary it was added as string

**Pull request merged**